File: output/ChatGPT/codenet/C++/Java/s669128523.java
import java.util.Scanner;

public class s669128523 {
    private static final int N = 100;
    private static final int WHITE = 0;
    private static final int GRAY = 1;
    private static final int BLACK = 2;

    private static int[][] M = new int[N][N];
    private static int[] color = new int[N];
    private static int[] d = new int[N];
    private static int[] f = new int[N];
    private static int tt;

    private static void dfs_visit(int u) {
        int v;
        color[u] = GRAY;
        d[u] = ++tt;
        for (v = 0; v < N; v++) {
            if (M[u][v] == 0)
                continue;
            if (color[v] == WHITE) {
                dfs_visit(v);
            }
        }
        color[u] = BLACK;
        f[u] = ++tt;
    }

    private static void dfs() {
        int u;
        for (int i = 0; i < N; i++) {
            color[i] = WHITE;
        }
        tt = 0;

        for (u = 0; u < N; u++) {
            if (color[u] == WHITE)
                dfs_visit(u);
        }

        for (int i = 0; i < N; i++)
            System.out.println((i + 1) + " " + d[i] + " " + f[i]);
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int n = input.nextInt();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++)
                M[i][j] = 0;
        }
        for (int i = 0; i < n; i++) {
            int u = input.nextInt() - 1;
            int k = input.nextInt();
            for (int j = 0; j < k; j++) {
                int v = input.nextInt() - 1;
                M[u][v] = 1;
            }
        }

        dfs();
    }
}

File: output/ChatGPT/codenet/C++/Python/s669128523.py
#depth first search
#not lie in stack

N=100
WHITE=0
GRAY=1
BLACK=2

M=[[0]*N for i in range(N)]
n=0
color=[0]*N
d=[0]*N
f=[0]*N
tt=0

def dfs_visit(u):
    global tt
    v=0
    color[u]=GRAY
    d[u]=tt+1
    while v<n:
        if M[u][v]==0:
            v+=1
            continue
        if color[v]==WHITE:
            dfs_visit(v)
        v+=1
    color[u]=BLACK
    tt+=1
    f[u]=tt

def dfs():
    global tt
    u=0
    for i in range(n):
        color[i]=WHITE
    tt=0
    while u<n:
        if color[u]==WHITE:
            dfs_visit(u)
        u+=1
    for i in range(n):
        print(i+1,d[i],f[i])

n=int(input())
for i in range(n):
    for j in range(n):
        M[i][j]=0
for i in range(n):
    u,k=map(int,input().split())
    u-=1
    for j in range(k):
        v=int(input())
        v-=1
        M[u][v]=1

dfs()